src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -463,7 +463,7 @@ public function next_tag( $query = null ) {
 					continue;
 				}
 
-				if ( ! parent::is_tag_closer() || $visit_closers ) {
+				if ( ! $this->is_tag_closer() || $visit_closers ) {
```

Comment:
@dmsnell This came up here: https://github.com/WordPress/wordpress-develop/pull/6348#discussion_r1624278322

I haven't been able to come up with a failing test case for it.